client/my-sites/plan-features-2023-grid/plan-comparison-grid.tsx

```diff
@@ -1,21 +1,193 @@
+`;
+
+type FeatureAvailabilityForPlansProps = {
+	displayedPlans?: Array< PlanProperties >;
+	featureSlug: string;
+	planFeatureMap: Record< string, Set< string > >;
+};
+
+const FeatureAvailabilityForPlans: React.FC< FeatureAvailabilityForPlansProps > = ( {
+	displayedPlans,
+	featureSlug,
+	planFeatureMap,
+} ) => {
+	return (
+		<>
+			{ ( displayedPlans ?? [] ).map( ( { planName } ) => {
+				if ( planFeatureMap[ planName ].has( featureSlug ) ) {
+					return (
+						<Cell
+							key={ planName }
+							className={ `plan-comparison-grid__plan ${ planName }` }
+							textAlign="center"
+						>
+							<Gridicon icon="checkmark" color="#0675C4" />
+						</Cell>
+					);
+				}
+				return (
+					<Cell
+						key={ planName }
+						className={ `plan-comparison-grid__plan ${ planName }` }
+						textAlign="center"
+					>
+						<Gridicon icon="minus-small" />
+					</Cell>
+				);
+			} ) }
+		</>
+	);
+};
+
+type PlanComparisonGridProps = {
+	planProperties?: Array< PlanProperties >;
+	intervalType: string;
+};
+
+export const PlanComparisonGrid: React.FC< PlanComparisonGridProps > = ( {
+	planProperties,
+	intervalType,
+} ) => {
 	const translate = useTranslate();
+	const planGroups = getPlanFeaturesGrouped();
+	const displayedPlansProperties = ( planProperties ?? [] ).filter(
+		( { planName } ) => ! ( planName === PLAN_ENTERPRISE_GRID_WPCOM )
+	);
+	const isMonthly = intervalType === 'monthly';
+	const planFeatureMap = useMemo( () => {
+		let previousPlan = null;
+		const featureMap: Record< string, Set< string > > = {};
+
+		for ( const plan of planProperties ?? [] ) {
+			const { planName } = plan;
+			const planObject = applyTestFiltersToPlansList( planName, undefined );
+			const wpcomFeatures = planObject.get2023PricingGridSignupWpcomFeatures?.() ?? [];
+			const jetpackFeatures = planObject.get2023PricingGridSignupJetpackFeatures?.() ?? [];
+			const annualOnlyFeatures = planObject.getAnnualPlansOnlyFeatures?.() ?? [];
+
+			let featuresAvailable = [ ...wpcomFeatures, ...jetpackFeatures ];
+			if ( isMonthly ) {
+				// Filter out features only available annually
+				featuresAvailable = featuresAvailable.filter(
+					( feature ) => ! annualOnlyFeatures.includes( feature )
+				);
+			}
+			featureMap[ planName ] = new Set( featuresAvailable );
+
+			// Add previous plan feature
+			if ( previousPlan !== null ) {
+				featureMap[ planName ] = new Set( [
+					...featureMap[ planName ],
+					...featureMap[ previousPlan ],
+				] );
+			}
+			previousPlan = planName;
+		}
+		return featureMap;
+	}, [ planProperties, isMonthly ] );
+
 	return (
 		<div className="plan-comparison-grid">
 			<PlanComparisonHeader className="wp-brand-font">
 				{ translate( 'Compare our plans and find yours' ) }
 			</PlanComparisonHeader>
+			<Grid>
+				<Row key="feature-group-title" className="plan-comparison-grid__plan-row">
+					<Cell key="feature-name" className="plan-comparison-grid__interval-toggle">
+						<BrowserRouter>
```

Comment:
I'm not sure if `BrowserRouter` can be used here. Can you please explain the usage?

Reply:
Deep in this component there is a usage of the Link component. Without the routing context the code crashes. 🤔

Reply:
This has been commented out for now.